extensions/test/NETCore.SetupTests/ConfigurationTests.cs

```diff
@@ -122,10 +122,10 @@ public void GetClientConfigSettingsTest()
             Assert.Equal(DefaultConfigurationMode.Standard, client.Config.DefaultConfigurationMode);
             Assert.Equal(RequestRetryMode.Adaptive, client.Config.RetryMode);
 
-            // verify S3 config specific settings are not configured
+            // verify S3 config specific settings are configured
             var clientConfig = client.Config as AmazonS3Config;
             Assert.NotNull(clientConfig);
-            Assert.False(clientConfig.ForcePathStyle);
+            Assert.True(clientConfig.ForcePathStyle);
```

Comment:
I changed this to `true` because the `GetClientConfigSettingsTest.json` file says it should be true. The test was saying it was false because it was call the `GetAWSOptions` option without the generic passing in the AmazonS3Config. I view that as a limitation of the old implementation as it makes more logical sense that if the user set the value it should be on the client config. With the new approach in this PR that removed the generic version the service specific settings are always set if they are set in appsettings.json and the service client config being created.

TLDR: The behavior got more accurate to the user's intent.